src/compiler/watchUtilities.ts

```diff
@@ -34,9 +34,20 @@ namespace ts {
         clearCache(): void;
     }
 
+    type Canonicalized = string & { __canonicalized: void };
+
     interface MutableFileSystemEntries {
         readonly files: string[];
         readonly directories: string[];
+        sortedAndCanonicalizedFiles?: undefined | SortedArray<Canonicalized>
+        sortedAndCanonicalizedDirectories?: undefined | SortedArray<Canonicalized>

```

Comment:
Silly nit but why is `undefined` first here? (Or present at all?)

Reply:
Putting it first was arbitrary, having it there at all was because they didn't start out optional.  Removed.